iceoryx_dust/test/moduletests/test_cxx_string_conversion.cpp

```diff
@@ -108,4 +110,618 @@ TYPED_TEST(StdString_test, StringOfSizeCapaToSTDStringConvResultsInSizeCapa)
+    EXPECT_THAT(sut.size(), Eq(7U));
+    EXPECT_THAT(std::memcmp(sut.c_str(), expectedString.c_str(), sut.size()), Eq(0));
+}
+
+TYPED_TEST(StdString_test, FindStdStringInEmptyStringFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "75e34d61-be16-4892-a931-96a96bc1e45f");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    string<STRINGCAP + 5U> testString("a");
+    auto res = this->testSubject.find(testString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    std::string testStdString = "a";
+    res = this->testSubject.find(testStdString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+/// @note template <typename T>
+/// iox::cxx::optional<uint64_t> find_first_of(const T& t, uint64_t pos = 0) const noexcept
+TYPED_TEST(StdString_test, FindFirstOfFailsForEmptyStdStringInEmptyString)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "207671e4-cef3-40d2-8984-e8ae5c2b42ec");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    string<STRINGCAP + 5U> testString;
+    auto res = this->testSubject.find_first_of(testString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    std::string testStdString;
+    res = this->testSubject.find_first_of(testStdString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+TYPED_TEST(StdString_test, FindFirstOfForStdStringInEmptyStringFails)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "cfd4a842-64e3-4a2c-afc9-d98f93f1f8f4");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    string<STRINGCAP + 5U> testString("a");
+    auto res = this->testSubject.find_first_of(testString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+
+    std::string testStdString = "a";
+    res = this->testSubject.find_first_of(testStdString);
+    EXPECT_THAT(res.has_value(), Eq(false));
+}
+
+TEST(String100, FindFirstOfForSTDStringInNotEmptyStringWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "1f44acab-aa37-4f45-a782-06ad02bd926b");
+    string<100U> testString("R2-D2");
+    std::string testStdString1 = "2";
+    auto res = testString.find_first_of(testStdString1);
+    ASSERT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(1U));
+
+    res = testString.find_first_of(testStdString1, 1U);
+    ASSERT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(1U));
+
+    res = testString.find_first_of(testStdString1, 2U);
+    ASSERT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(4U));
+
+    std::string testStdString2 = "D3R";
+    res = testString.find_first_of(testStdString2);
+    ASSERT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(0U));
+
+    res = testString.find_first_of(testStdString2, 1U);
+    ASSERT_THAT(res.has_value(), Eq(true));
+    EXPECT_THAT(res.value(), Eq(3U));
+}
+
+TEST(String100, FindFirstOfForNotIncludedSTDStringFails)
```

Comment:
This is also something for a follow up together with `test_vocabulary_string.cpp`. It is weird to have typed tests intermixed with non typed tests